src/include/sof/math/numbers.h

```diff
@@ -35,8 +35,16 @@
 
 #include <stdint.h>
 
-#define MIN(a, b) (((a) < (b)) ? (a) : (b))
-#define MAX(a, b) (((a) > (b)) ? (a) : (b))
+#define MIN(a, b) ({		\
```

Comment:
I'm curious, why is this better?

Reply:
if you have MIN(x++, y), then with the first version you'd get
```
(x++ > y) ? x++ : y;
```
i.e. you'd have x incremented twice. With the second version you'd correctly have it incremented once. Same for any MIN(f(x), y) - f(x) would be called twice with the first version.